src/js/component.js

```diff
@@ -359,19 +365,20 @@ vjs.Component.prototype.getChild = function(name){
  * @return {vjs.Component} The child component (created by this process if a string was used)
  * @suppress {accessControls|checkRegExp|checkTypes|checkVars|const|constantProperty|deprecated|duplicate|es5Strict|fileoverviewTags|globalThis|invalidCasts|missingProperties|nonStandardJsDocs|strictModuleDepCheck|undefinedNames|undefinedVars|unknownDefines|uselessCode|visibility}
  */
-vjs.Component.prototype.addChild = function(child, options){
-  var component, componentClass, componentName;
-
-  // If child is a string, create new component with options
+Component.prototype.addChild = function(child, options){
+  let component, componentName;
```

Comment:
I see the value of marking somehow long lived variables. But if you enforce the use of 'let' you are basically sending a very clear message. Variables should be block scoped whenever possible. People is used to 'var' and they will unconsciously use it whenever possible but with 'let' the will be forced to think on the visibility of the variable.  I also makes it easier. You don't have to think do I use let or var everytime you create a variable.